src/index.ts

```diff
@@ -1,46 +1,43 @@
-import { Response, Headers as FetchHeaders } from 'node-fetch';
-import FormData from 'form-data';
+import { Response, Request, Headers as FetchHeaders, RequestInfo, RequestInit } from 'node-fetch';
 import { AxiosInstance, AxiosRequestConfig } from './types';
+import FormData from 'form-data';
 
-export interface FetchInit extends Record<string, any> {
-  headers?: Record<string, string>;
-  method?: AxiosRequestConfig['method'];
-  body?: FormData | any;
-  extra?: any;
-}
-
-export type AxiosTransformer = (config: AxiosRequestConfig, input: string | undefined, init: FetchInit) => AxiosRequestConfig;
+export type AxiosTransformer = (config: AxiosRequestConfig, input: RequestInfo, init?: RequestInit) => AxiosRequestConfig;
 
-export type AxiosFetch = (input?: string, init?: FetchInit) => Promise<Response>;
+export type AxiosFetch = (input: RequestInfo, init?: RequestInit) => Promise<Response>;
 
 /**
  * A Fetch WebAPI implementation based on the Axios client
+ *
+ * @param axios
+ * @param transformer Convert the `fetch` style arguments into a Axios style config
+ * @param input
+ * @param init
  */
 async function axiosFetch (
   axios: AxiosInstance,
-  // Convert the `fetch` style arguments into a Axios style config
-  transformer?: AxiosTransformer,
-  input?: string,
-  init: FetchInit = {}
+  transformer: AxiosTransformer | undefined,
+  input: RequestInfo,
+  init?: RequestInit
 ) {
-  const rawHeaders: Record<string, string> = init.headers || {};
-  const lowerCasedHeaders = Object.keys(rawHeaders)
-    .reduce<Record<string, string>>(
-      (acc, key) => {
-        acc[key.toLowerCase()] = rawHeaders[key];
-        return acc;
-      },
-      {}
-    );
+  // Request class handles for us all the input normalisation
+  const request = new Request(input, init);
+  const lowerCasedHeaders: Record<string, string> = {};
+
+  for (const entry of request.headers.entries()) {
```

Comment:
Is there a functional change between this new version and the old `Object.keys` version? If not, can you restore the old version to minimize the diff?  If there is, can you add a testcase to verify the new behavior?

Reply:
I overlooked the fact that `headers` is not a different object type. Your implemenation is fine